app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentEffectHandler.kt

```diff
@@ -107,12 +107,7 @@ class ScheduleAppointmentEffectHandler @AssistedInject constructor(
   private fun loadPatientDefaulterStatus(): ObservableTransformer<LoadPatientDefaulterStatus, ScheduleAppointmentEvent> {
     return ObservableTransformer { effects ->
       effects
-          // TODO (vs) 20/05/20: Make this a synchronous call
-          .switchMapSingle {
-            patientRepository
-                .isPatientDefaulter(it.patientUuid)
-                .firstOrError()
-          }
+          .map { patientRepository.isPatientDefaulter(it.patientUuid) }
```

Comment:
Maybe we can use a single `map` function, wdyt?

```suggestion
effects
    .map {
      val isPatientDefaulter = patientRepository.isPatientDefaulter(it.patientUuid)
      PatientDefaulterStatusLoaded(isPatientDefaulter)
    }
```

Reply:
I could do this, will make the change.

Reply:
Done